packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+import 'dart:typed_data';
+import 'dart:ui'  show hashValues;
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+
+import 'binding.dart';
+import 'debug.dart';
+import 'image_provider.dart' as image_provider;
+import 'image_stream.dart';
+
+/// The dart:io implemenation of [image_provider.NetworkImage].
+class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkImage> implements image_provider.NetworkImage {
+  /// Creates an object that fetches the image at the given URL.
+  ///
+  /// The arguments must not be null.
```

Comment:
What about the headers argument? (It should either be excluded in this sentence or there should be a non-null assert)

Reply:
Update doc comment to state that only scale and uri need be non-null